oslabs-beta/reactime
# Initial refactor of npm package to support hooks state and improve code reusability.

## Description:
- Refactored npm package to retrieve state changes in hooks components and send them to Chrome extension. 

## Changes I Made:
- Refactored linkFiber.js to grab Fiber root through React DevTools instead of npm package. First step in transition out of using an npm package.
- Fixed errors with hooks components states not being properly passed to extension.
- Implemented an indexed structure in masterState.js to keep setState() and dispatch() functions for changing component states, and added indexes to snapshots to track them.
- Refactored tree.js to simplify creation of state snapshots, and solved problems with creating streamable copies of functions and circular objects.
- Refactored timeJump.js to use new index structure in masterState.js.

## How to Test:
- Extension works the same as previous versions, but now you should see hooks components states in tree chart. 
- Testing is needed for apps using setState components that load and unload the components.
